src/coord/src/coord.rs

```diff
@@ -332,6 +334,49 @@ fn concretize_replica_config(
     Ok(config)
 }
 
+/// Holds tables needing advancement.
+struct AdvanceTables<T> {
+    map: HashMap<GlobalId, T>,
+}
+
+impl<T: CoordTimestamp> AdvanceTables<T> {
+    fn new() -> Self {
+        Self {
+            map: HashMap::new(),
+        }
+    }
+
+    // Insert an id that should be advanced to ts. If an existing and earlier entry
+    // exists, update it to ts.
+    fn insert(&mut self, id: GlobalId, ts: T) {
+        match self.map.entry(id) {
+            Entry::Occupied(mut entry) => {
+                if entry.get().less_than(&ts) {
+                    entry.insert(ts);
+                }
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(ts);
+            }
+        }
+    }
+
+    // Returns the next table to advance. Blocks forever if there are none.
+    async fn recv(&mut self) -> (GlobalId, T) {
+        // TODO(mjibson): In order to prevent starvation, we need to make this random
+        // or somehow fair. As is, the iteration order is arbitrary, so it is possible
+        // some things never get visited in busy coordinators.
+        let key: Option<GlobalId> = self.map.keys().next().cloned();
+        match key {
+            None => futures::future::pending().await,
```

Comment:
I would put a comment on this function explaining that `recv` will never finish if there is nothing to do, and that this is required by the current coord implementation and nothing else

Reply:
The never finish comment already exists. I was going to add some words talking about its use in coord but I thought it was clearer just to not instead?